hips/hip-9997.md

```diff
@@ -0,0 +1,197 @@
+
+## Backwards compatibility
+
+Since the existing feature set is currently experimental, there will be no promise of backwards compatibility with prior OCI support (sorry!)
+
+## Security implications
+
+Registry authentication introduces a new attack vector. Upon running `helm registry login`, these credentials are stored in an unencrypted JSON file. With that said, the underlying code is leveraging docker/cli, which has the ability to store these credentials into the system keychain instead.
+
+So, while a new attack vector has been introduced, it should not be considered any less safe than using `docker login`.
+
+Note: this is already implemented in the current OCI feature set.
+
+## How to teach this
+
+Following the implementation of all facets of this HIP, extensive documentation should be added to the Helm website on how to leverage OCI.
+
+Additionally, sites containing community charts, such as [Artifact Hub](https://artifacthub.io/), should enable providers to distribute charts over OCI, and display friendly copy-paste instructions for how to download an OCI-based chart.
+
+## Reference implementation
+
+Please see [the Helm documentation](https://helm.sh/docs/topics/registries/) for existing OCI support.
+
+## Rejected ideas
+
+N/A
+
+## Open issues
+
+The issues below are still unresolved.
+
+### What is the correct media type to use for the chart content layer?
+
+The media type for the chart content layer is currently `application/tar+gzip`, however it has been pointed out that [this is not an official type in the IANA database](https://www.iana.org/assignments/media-types/media-types.xhtml).
+
+Either a.) we go through with registering the media type through IANA, or b.) use a unique media type. One alternative could be `application/vnd.cncf.helm.chart.content.layer.v1+tar`.
+
+The only real issue with option b is that the chart package is not technically a unique type. As in, a registry vendor could treat it as a typical gzipped tarball.
+
+### What is the correct media type to use for the provenance file layer?
+
+Again, there doesn't appear to be an IANA media type for this type of file. Additionally, this file format appears unique to Helm.
+
+In this case, it seems the only option is to use a custom media type such as `application/vnd.cncf.helm.chart.provenance.layer.v1`.
+
+### Will I be able to use custom tags such as "latest"?
+
+The implications of this are not yet fully understood. For now, it will be strictly enforced that the tag on the reference will match the version found in `Chart.yaml`. This functionality may be added at a later time.
+
+### Will I be able to pull a chart using the `:<tag>` syntax?
+
+It's unclear if users will be able to pull charts with a specific version using the `:<tag>` syntax (vs. `--version <version>`). This seems relatively harmless to allow both, but this may have implications on dependencies and other areas.
```

Comment:
I think given that the proposed workflow is `helm package` && `helm push`, where the tag and the chart version match 1:1, this should be moved under the "Rejected Ideas" section along with the previous question on using the "latest" tag.

Once the OCI proposal moves forward and becomes stable, the community can propose future HIPs to try and tackle this scenario. That way we can discover

- what use cases exist for arbitrary tag names
- how they plan to address the concerns with tag names not matching with the chart version
- how is this useful to the helm community

Reply:
done